ledger/store/src/helpers/rocksdb/internal/map.rs

```diff
@@ -109,55 +111,55 @@ impl<
     ///
     fn atomic_checkpoint(&self) {
         // Push the current length of the atomic batch to the checkpoint stack.
-        self.checkpoint.lock().push(self.atomic_batch.lock().len());
+        self.checkpoints.lock().push(self.atomic_batch.lock().len());
     }
 
     ///
     /// Removes the latest atomic checkpoint.
     ///
     fn clear_latest_checkpoint(&self) {
         // Removes the latest checkpoint.
-        let _ = self.checkpoint.lock().pop();
+        let _ = self.checkpoints.lock().pop();
     }
 
     ///
     /// Removes all pending operations to the last `atomic_checkpoint`
     /// (or to `start_atomic` if no checkpoints have been created).
     ///
     fn atomic_rewind(&self) {
-        // Acquire the write lock on the atomic batch.
-        let mut atomic_batch = self.atomic_batch.lock();
-
         // Retrieve the last checkpoint.
-        let checkpoint = self.checkpoint.lock().pop().unwrap_or(0);
-
+        let checkpoint = self.checkpoints.lock().pop().unwrap_or(0);
         // Remove all operations after the checkpoint.
-        atomic_batch.truncate(checkpoint);
+        self.atomic_batch.lock().truncate(checkpoint);
     }
 
     ///
     /// Aborts the current atomic operation.
     ///
     fn abort_atomic(&self) {
         // Clear the atomic batch.
-        *self.atomic_batch.lock() = Default::default();
+        self.atomic_batch.lock().clear();
```

Comment:
Just curious: Is there a benefit to using `clear` vs reassigning to `Default::default()`?

Reply:
yes,;this reuses the previous allocation, which is very likely of an appropriate size